library/gdalframework/src/main/java/it/geosolutions/imageio/gdalframework/GDALUtilities.java

```diff
@@ -764,6 +767,81 @@ public static ColorModel buildColorModel(final SampleModel sampleModel) {
         return colorModel;
     }
 
+    /**
+     * Extracts the bands from the provided {@link ImageReadParam},
+     * with EnhancedImageReadParam.getBands() taking precedence
+     * over ImageReadParam.getSourceBands() and overriding it.
+     *
+     * @param imageReadParam
+     * @return
+     */
+    public static int[] extractBands(ImageReadParam imageReadParam) {
+        int[] srcBands = imageReadParam.getSourceBands();
+        if (imageReadParam instanceof EnhancedImageReadParam) {
+            // Let's check if the EnhancedImageReadParam has been set and it contains
+            // a bands array. In that case, this will take precedence over the srcBands
+            final EnhancedImageReadParam eparam = (EnhancedImageReadParam) imageReadParam;
+            if (eparam.getBands() != null) {
+                srcBands = eparam.getBands();
+                imageReadParam.setSourceBands(srcBands);

```

Comment:
Having a side effect here... is it necessary? It's not documented in the javadoc.